artifactory/commands/transferfiles/utils.go

```diff
@@ -119,21 +121,29 @@ func pollUploads(phaseBase *phaseBase, srcUpService *srcUserPluginService, uploa
 		if progressbar != nil {
 			progressbar.SetRunningThreads(curProcessedUploadChunks)
 		}
-		curTokensBatch.fillTokensBatch(uploadTokensChan)
-
-		if len(curTokensBatch.UuidTokens) == 0 {
+		// Each uploading thread receive a token from the source via the uploadTokensChan, so this go routine can poll on it's status
+		fillTokensBatch(awaitingStatusChunksSet, uploadTokensChan)
+		// awaitingStatusChunksSet is used to keep all the uploaded chunks tokens in order to request their upload status from the source
+		// ChunksToDelete is used to notify the source that these chunks can be deleted from the source's status map
+		// After we receive 'DONE', we inform the source that the 'DONE' message has been received, and it no longer has to keep those chunk uuids.
+		// When both ChunksToDelete and awaitingStatusChunksSet length is zero means that all the tokens has been uploaded,
+		// we received 'DONE' for all of them, and we notified the source that they can be deleted from the memory
```

Comment:
Missing period at the end of the sentence.

Reply:
done